src/coreclr/src/jit/gentree.h

```diff
@@ -5991,48 +5991,151 @@ struct GenTreePhiArg : public GenTreeLclVarCommon
 
 struct GenTreePutArgStk : public GenTreeUnOp
 {
+private:
+    unsigned m_byteOffset;
+#ifdef FEATURE_PUT_STRUCT_ARG_STK
+    unsigned m_byteSize; // The number of bytes that this argument is occupying on the stack with padding.
+#endif
+
+public:
+#if defined(DEBUG_ARG_SLOTS)
     unsigned gtSlotNum; // Slot number of the argument to be passed on stack
+#if defined(FEATURE_PUT_STRUCT_ARG_STK)
+    unsigned gtNumSlots; // Number of slots for the argument to be passed on stack
+#endif
+#endif
+
 #if defined(UNIX_X86_ABI)
     unsigned gtPadAlign; // Number of padding slots for stack alignment
 #endif
+#if defined(DEBUG) || defined(UNIX_X86_ABI)
+    GenTreeCall* gtCall; // the call node to which this argument belongs
+#endif
 
-    // Don't let clang-format mess with the GenTreePutArgStk constructor.
-    // clang-format off
+#if FEATURE_FASTTAILCALL
+
+    bool gtPutInIncomingArgArea; // Whether this arg needs to be placed in incoming arg area.
+                                 // By default this is false and will be placed in out-going arg area.
+                                 // Fast tail calls set this to true.
+                                 // In future if we need to add more such bool fields consider bit fields.
+#endif
+
+#ifdef FEATURE_PUT_STRUCT_ARG_STK
+    // Instruction selection: during codegen time, what code sequence we will be using
+    // to encode this operation.
+    // TODO-Throughput: The following information should be obtained from the child
+    // block node.
+
+    enum class Kind : __int8{
+        Invalid, RepInstr, Unroll, Push, PushAllSlots,
+    };
+    Kind gtPutArgStkKind;
+#endif
 
+#if defined(DEBUG_ARG_SLOTS) && defined(FEATURE_PUT_STRUCT_ARG_STK)
```

Comment:
I like more how you did the `ifdef`-magic for `GenTreePutArgSplit` class.
Do you think the same style can be applied here?

I mean have the arguments guarded by two nested ifdefs instead of flattening them into 4 separate cases?
```c++
#if defined(FEATURE_PUT_STRUCT_ARG_STK)
                       unsigned stackByteSize,
#endif
#if defined(DEBUG_ARG_SLOTS)
                       unsigned slotNum,
#if defined(FEATURE_PUT_STRUCT_ARG_STK)
                       unsigned numSlots,
#endif
#endif
```